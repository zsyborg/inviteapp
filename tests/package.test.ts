import * as fsAsync from 'fs/promises';
import * as fs from 'fs';
import * as path from 'path';

const BUILD_PATH = path.resolve(__dirname, '../lib');
const COMMON_JS_MODULE = 'react-telegram-web-app.cjs';

const walk = async (dirPath: string): Promise<any> =>
  Promise.all(
    await fsAsync.readdir(dirPath, { withFileTypes: true }).then(entries =>
      entries.map(entry => {
        const childPath = path.join(dirPath, entry.name);
        return entry.isDirectory()
          ? walk(childPath)
          : childPath.replace(BUILD_PATH, '');
      }),
    ),
  );

describe('Package /lib', () => {
  beforeEach(() => {
    jest.resetModules();
  });

  it('should have correct /lib structure', async () => {
    expect(await walk(BUILD_PATH)).toMatchSnapshot();
  });

  it('should have exports from modules', () => {
    const indexModule = require(path.join(BUILD_PATH, COMMON_JS_MODULE));

    expect(indexModule).toMatchSnapshot();
  });
});

describe('Public API documentation', () => {
  it('should have describe in README.md', () => {
    const indexModule = Object.keys(
      require(path.join(BUILD_PATH, COMMON_JS_MODULE)),
    );

    const mdFile = fs.readFileSync(
      path.resolve(__dirname, '../README.md'),
      'utf8',
    );

    expect(
      indexModule
        .map(name => `[${name}](./docs/README.md#${name.toLowerCase()})`)
        .filter(name => !mdFile.includes(name)),
    ).toStrictEqual([]);
  });
});
