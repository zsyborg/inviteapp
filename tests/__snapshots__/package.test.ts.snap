// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Package /lib should have correct /lib structure 1`] = `
[
  "/BackButton.d.ts",
  "/MainButton.d.ts",
  "/WebAppProvider.d.ts",
  [
    "/core/context.d.ts",
    "/core/index.d.ts",
    [
      "/core/twa-types/WebApp.d.ts",
      "/core/twa-types/WebAppVersion_6.1.d.ts",
      "/core/twa-types/WebAppVersion_6.2.d.ts",
      "/core/twa-types/WebAppVersion_6.4.d.ts",
      "/core/twa-types/WebAppVersion_6.7.d.ts",
      "/core/twa-types/WebAppVersion_6.9.d.ts",
      "/core/twa-types/index.d.ts",
    ],
    "/core/useAsyncMode.d.ts",
    "/core/useSmoothButtonsTransition.d.ts",
    "/core/useWebApp.d.ts",
  ],
  "/index.d.ts",
  "/react-telegram-web-app.cjs",
  "/react-telegram-web-app.modern.js",
  "/react-telegram-web-app.module.js",
  "/react-telegram-web-app.umd.js",
  "/useCloudStorage.d.ts",
  "/useExpand.d.ts",
  "/useHapticFeedback.d.ts",
  "/useInitData.d.ts",
  "/useReadTextFromClipboard.d.ts",
  "/useScanQrPopup.d.ts",
  "/useShowPopup.d.ts",
  "/useSwitchInlineQuery.d.ts",
  "/useThemeParams.d.ts",
  "/useWebApp.d.ts",
]
`;

exports[`Package /lib should have exports from modules 1`] = `
{
  "BackButton": [Function],
  "MainButton": [Function],
  "WebAppProvider": [Function],
  "useCloudStorage": [Function],
  "useExpand": [Function],
  "useHapticFeedback": [Function],
  "useInitData": [Function],
  "useReadTextFromClipboard": [Function],
  "useScanQrPopup": [Function],
  "useShowPopup": [Function],
  "useSwitchInlineQuery": [Function],
  "useThemeParams": [Function],
  "useWebApp": [Function],
}
`;
